mathesar_ui/src/layouts/LayoutWithHeader.svelte

```diff
@@ -1,30 +1,67 @@
 <script lang="ts">
   import AppHeader from '@mathesar/components/AppHeader.svelte';
+  import LiveDemoBanner from '@mathesar/components/LiveDemoBanner.svelte';
 
   export let fitViewport = false;
+  export let restrictWidth = false;
+  export let cssVariables: Record<string, string> | undefined = undefined;
+
+  $: style = cssVariables
+    ? Object.entries(cssVariables)
+        .filter((val) => val[0].indexOf('--') === 0)
+        .map((entry) => `${entry[0]}: ${entry[1]}`)
+        .join(';')
+    : undefined;
```

Comment:
Wow, you _really_ don't like style props, huh?

I'll admit that the wrapper element is a little weird. Sometimes it gets in the way if I'm trying to use child selectors, but I've usually been able to find better ways around that problem in those cases anyway. I don't really care too much though. I'm fine with adopting this `cssVariables` approach. I do have some thoughts about this code:

- It would be nice to extract this into a utility function so that we can easily employ this pattern consistently in other places.
- It would be nice to add a trailing semicolon so that the style string can be joined with others if needed.